test/integration_multithread_test.py

```diff
@@ -0,0 +1,82 @@
+from os import system
+from threading import Thread
+import socket
+import time
+import sys
+import subprocess
+
+
+def client_thread_paused(args):
+	TCP_IP = '127.0.0.1'
+	TCP_PORT = 2020
+	BUFFER_SIZE = 1024
+	MESSAGE = "GET /echo HTTP/1.1"
+
+	s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+	s.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, 1)
+	s.connect(('localhost', TCP_PORT))
+	s.send(MESSAGE)
+	time.sleep(5)
+	s.send("\r\n\r\n")
+	data = s.recv(BUFFER_SIZE)
+	s.close()
+
+	if (data != "HTTP/1.1 200 OK\r\nContent-Length: 18\r\nContent-Type: text/plain\r\n\r\nGET /echo HTTP/1.1\r\n\r\n"):
+	    	print "Failure response: ", repr(data)
+                error_code[0] = 1
+		return
+
+	error_code[0] = 0
+
+def client_thread(args):
```

Comment:
Aside from the sleep and message portion, these functions are identical. What do you think about deleting one of the two and just passing in the message to each. The main thing we need to check for in the response is 200OK which could be shared among the two requests. If anything you can have the function return the data and check it in main so you dont have to copy paste the function code.